### ISSUE TITLE
Generate Dependent Benefits Letter Requirements from test user flow

### ISSUE THREAD

**Issue by joeniquette** [department-of-veterans-affairs/va.gov-team#25826]
**Goal:**
dig into how this function works on the legacy site IOT develop requirements of what user data is required to generate the letter. Then determine which of the identity attributes we already have access to (from where exactly) and which are missing (status on getting them).

**Steps to test this feature in eBenefits right now, you can do the following:**
1. Browse to the eBenefits PINT environment Mock-CSP login page: https://pint.ebenefits.va.gov/domain1/mock-csp/index.jsp
           - This is the EVSS environment that would be used from VA.gov staging
3. Choose one of these dependent test users from the “Select Account” drop down menu:
4. Pena, Roberta
5. Wilson, Leona [ does not seem to be working at the moment, looks like a BGS partner issue ]
6. Click the “Send to eBenefits” button
           - Note the above is all associated with using a test identity to get into eBenefits, not part of the normal user experience
8. From the eBenefits dashboard, click the “Letter Generator” link under “Popular Items” on the right side of the screen
9. Click the link for the “Benefit Summary – Spouse or Dependent Benefits” from the list of available letters
10. Optional: adjust which information will be included in letter with the available checkboxes
11. Click the “Generate Benefit Summary Letter” button at the bottom of the page
12. A new browser tab will open and display the PDF letter generated for you (sample attached).

[https://app.zenhub.com/files/133843125/05196014-1002-4b4f-af49-a8b7dabb0434/download](https://app.zenhub.com/files/133843125/05196014-1002-4b4f-af49-a8b7dabb0434/download)

After we complete our part of the puzzle Matt Self, Jason Wolf, and Samara Strauss/Kevin Hoffman will decide on:
- how to parse the user attributes we provide/generate the login in code
- create and decide on the frontend content to be user for the generation of the letter
- generate the letter and present to dependent

**Comment by bosawt** [comment]
Here is the summary from looking through relevant documents:

# Dependent Benefits Letter Requirements
### Summary
* BSL
	* Benefits Summary Letter
### Requirements to show Letter
* User is a (surviving) spouse or dependent
* User is LOA3
### Letter Abilities
* Select or unselect certain elements
	* Specifically calls out:
		* Ability to display all the periods of service in the Military Service Information section or not
		* Choose whether to display the monthly award amount, and user award effective date
* Generation of letter
* Ability to view letter in read only format
* Ability to print the letter
* Ability to save the letter
### Notes
* If user is a Veteran and a dependent, they should be able to generate both the Veteran and non-Veteran Letters
* There are a few ‘future enhancement’ options that appear to not be currently implemented but possible we could
	* If Veteran is 50% or greater, can be informed of the ‘no cost’ health care
	* There’s a section about ‘Do you Have Questions or Need Assistance’, apparently it directs to IRIS, suggestion was that it not direct to www.va.gov
	* Recommendation to add the Health Benefits link, va.gov/healthbenefits, to the letter in the ‘Do You Have Questions or Need Assistance’ section
### Template
* Header
	* All header fields are required
	* Date Letter Generated
		* Format: Month DD, YYYY
	* In Reply Refer To
		* Apparently SSN, or some alternate number
			* Add digits up to 8 numbers if less than 8 numbers
			* Show 8 digits as XX-XXX-XXX
			* Show 9 digits as XXX-XX-XXXX
				* Mask all except last 4 digits
	* User First Name
	* User Last Name
	* User Address
		* User Mailing Address Type
			* Domestic
				* Street Line 1
				* Street Line 2
				* Street Line 3
				* City
				* State
				* Zip Code
			* International
				* Street Line 1
				* Street Line 2
				* Street Line 3
				* City
				* Country
			* Military Overseas
				* Street Line 1
				* Street Line 2
				* Street Line 3
				* P.O.
				* Postal Code
				* Zip Code
	* Salutation
		* If Gender = Male
			* Use Mr.
		* If Gender = Female
			* Use Ms.
		* Else
			* Use First Name
* Personal Claim Information
	* This is a required field
	* VA Claim/File Number
		* Apparently SSN, or some alternate number
			* Add digits up to 8 numbers if less than 8 numbers
			* Show 8 digits as XX-XXX-XXX
			* Show 9 digits as XXX-XX-XXXX
				* Mask all except last 4 digits
* Military Information
	* These are all conditional fields
		* Only display these if record is VERIFIED in BIRLS, when VADS/VERIFIED field is Y
			* Display the last 3 records in BIRLS
		* Also, user must select include Military Service History section in letter options
	* Branch of Service
	* Character of Service
		* If HVA, DVA, OTH
			* Display Other Than Honorable
		* If UHC
			* Display Under Honorable Conditions
		* If GEN
			* Display General
		* If UNC
			* Display Uncharacterized
		* If UEL
			* Display Uncharacterized Entry Level
		* If DIS
			* Display Dishonorable
	* Entered Active Duty
		* Month DD, YYYY
	* Released/Discharged
		* Month DD, YYYY
* VA Benefits Information
	* These are all conditional fields
	* You are in receipt of:
		* Shown if user selects to include You are in Receipt of in the letter
		* Display either:
			* Dependency and Indemnity Compensation
			* Survivors Pension
	* Monthly award amount
		* Shown if user has selected to include Monthly Award Amount, and if there is a running award, comp or pension
		* This is a number value representing a dollar amount
	* User award effective date
		*  Shown if user has selected to include User Award Effective Date, and if there is a running award, comp or pension, and monthly award amount is not null
		* This is latest award effective date that is not in the future
		* Month DD, YYYY
	* Veteran Died as a result of service-connected disability
		* Show if user has selected Veteran Died as a Result of service-connected Disability
		* If dependent is in receipt of DIC, then display Yes
		* If not in receipt of DIC, display No
		* This information is in Corp DB, in the Rating Award Detail table
	* Considered totally and permanently disabled
		* This might actually be always shown, there’s a bit of a contradiction here
		* Show if user has selected Considered Totally and Permanently Disabled
		* If user is awarded CH35, then Yes, else No

**Comment by bosawt** [comment]
TLDR, we already have all the user profile info necessary for the letters, here's the other information we need:
VA Claim/File Number - this appears to mostly be SSN, but sometimes not? Still not completely sure what this is
Birls records (we should have birls_id from the MPI lookup, so just need to retrieve those records when generating the letter)
CH35 status - this appears to show whether or not the veteran is disabled
Status of DIC (whatever that is) - is apparantly from Corp DB - Rating Award Detail table, should be able to access this through BGS integrations
I'm not sure where the monthly award amount or user award effective date comes from, this is apparently either dependency and indemnity compensation or survivors pension, I assume these are sourced from Corp DB though

**Comment by joeniquette** [comment]
I am drafting the requirements to pass onto ebenefits for this. Will post on 6/28.

**Comment by joeniquette** [comment]
It was determined all user fields are going to be present, no additional asks required of MPI other than what was requested already.

**Comment by bosawt** [comment]
Here's an example of how to cURL on staging:

`curl --insecure -X POST \
-H 'va_eauth_csid: DSLogon' \
-H 'va_eauth_authenticationmethod: DSLogon' \
-H 'va_eauth_assurancelevel: 3' \
-H 'va_eauth_issueinstant: 2021-07-01T17:10:48Z' \
-H 'va_eauth_dodedipnid: 1013590059' \
-H 'va_eauth_birlsfilenumber: 796104437' \
-H 'va_eauth_pid: 13367440' \
-H 'va_eauth_pnid: 796104437' \
-H 'va_eauth_birthdate: 1950-10-04T00:00:00+00:00' \
-H 'va_eauth_authorization: {"authorizationResponse":{"id":796104437,"idType":"SSN","edi":1013590059,"firstName":"MARK","middleName":"","lastName":"WEBB","gender":"MALE","birthDate":"1950-10-04T00:00:00-07:00","deceased":false,"status":"SPONSOR","personnel":{"category":"RETIRED_MILITARY","entitlementCondition":"00","organization":25,"rank":"MCPO","serviceBranchClassification":"C"},"benefit":25,"headOfFamily":""}}' \
-H "Content-type: application/json" \
-H "accept: application/json" \
-d '{"militaryService":true,"monthlyAward":true,"serviceConnectedEvaluation":true,"chapter35Eligibility":true,"survivorsAward":true,"deathResultOfDisability":true}' \
'https://internal-dsva-vagov-staging-fwdproxy-1821450725.us-gov-west-1.elb.amazonaws.com:4431/wss-lettergenerator-services-web/rest/letters/v1/benefit_summary/generate'`
